lib/bolt/bolt_option_parser.rb

```diff
@@ -6,6 +6,76 @@
 
 module Bolt
   class BoltOptionParser < OptionParser
+    OPTIONS = { inventory: %w[nodes targets query rerun description],
+                authentication: %w[user password private-key host-key-check ssl ssl-verify],
+                escalation: %w[run-as sudo-password],
+                run_context: %w[concurrency inventoryfile save-rerun],
+                global_config_setters: %w[modulepath boltdir configfile],
+                transports: %w[transport connect-timeout tty tmpdir],
+                display: %w[format color verbose trace],
+                global: %w[help version debug] }.freeze
+
+    ACTION_OPTS = OPTIONS.values.flatten.freeze
+
+    def get_help_text(subcommand, action = nil)
+      case subcommand
+      when 'apply'
+        { flags: ACTION_OPTS + ['noop', 'execute', 'compile-concurrency'],
+          banner: APPLY_HELP }
+      when 'command'
+        { flags: ACTION_OPTS,
```

Comment:
I dont think run command uses a `tmpdir`. Not a huge deal, i think it will just be ignored. Other than that looks awesome!